packages/react-scripts/template/src/service-worker-registration.js

```diff
@@ -0,0 +1,38 @@
+export function register() {
+  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
```

Comment:
Yeah... I don't know much about Jest, and have yet to find a good way to create a mock environment for testing service worker-specific code outside of a real browser.

I'm hoping that some resolution would come out of those issues you reference. Is there anything that you'd suggest adding to this PR to work around it in the meantime?